File: rust/src/models/plan_version_summary.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::plan_version_id::PlanVersionId;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct PlanVersionSummary {
    pub created_at: String,

    pub currency: String,

    pub id: PlanVersionId,

    pub is_draft: bool,

    pub version: i32,
}

impl PlanVersionSummary {
    pub fn new(
        created_at: String,
        currency: String,
        id: PlanVersionId,
        is_draft: bool,
        version: i32,
    ) -> Self {
        Self {
            created_at,
            currency,
            id,
            is_draft,
            version,
        }
    }
}

File: typescript/src/models/planVersionSummary.ts
// this file is @generated
import { parseDateTime } from "../datetime";
import { type PlanVersionId, PlanVersionIdSerializer } from "./planVersionId";

export interface PlanVersionSummary {
  createdAt: Date;

  currency: string;

  id: PlanVersionId;

  isDraft: boolean;

  version: number;
}

export const PlanVersionSummarySerializer = {
  _fromJsonObject(object: any): PlanVersionSummary {
    return {
      createdAt: parseDateTime(object["created_at"]),
      currency: object["currency"],
      id: PlanVersionIdSerializer._fromJsonObject(object["id"]),
      isDraft: object["is_draft"],
      version: object["version"],
    };
  },

  _toJsonObject(self: PlanVersionSummary): any {
    return {
      created_at: self.createdAt,
      currency: self.currency,
      id: PlanVersionIdSerializer._toJsonObject(self.id),
      is_draft: self.isDraft,
      version: self.version,
    };
  },
};
